relayer/src/chain/cosmos/query/denom_trace.rs

```diff
@@ -0,0 +1,33 @@
+use http::uri::Uri;
+
+use ibc_proto::ibc::applications::transfer::v1::{
+    query_client::QueryClient, QueryDenomTraceRequest,
+};
+
+use crate::{denom::DenomTrace, error::Error};
+
+// Uses the GRPC client to retrieve the denom trace for a specific hash
+pub async fn query_denom_trace(grpc_address: &Uri, hash: &str) -> Result<DenomTrace, Error> {
+    let mut client = QueryClient::connect(grpc_address.clone())
+        .await
+        .map_err(Error::grpc_transport)?;
+
+    let request = tonic::Request::new(QueryDenomTraceRequest {
+        hash: hash.to_string(),
```

Comment:
`.to_owned()` is faster than going through the formatting framework.

Reply:
The whole crate is unnecessarily `pub`-happy, but if we threat this as an internal function and don't care about API nicety, the `hash` parameter of the function could be owned to begin with. It's passed as owned into the calling function, too.

Reply:
I see thank you. I changed the to_string calls to to_owned.

Reply:
Good. And the reference parameter is not worth bothering about, because the string has to be cloned also for the error case.